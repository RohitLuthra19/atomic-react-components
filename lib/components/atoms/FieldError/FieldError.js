// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import Para from '../Para';

import styles from './FieldError.style';

type Props = {
  className?: string,
  children: Node,
  dataSlnmId?: string,
  inheritedStyles?: string,
};

const FieldError = ({
  className,
  children,
  dataSlnmId,
  inheritedStyles,
  ...others
}: Props): Node => (
  <Para
    className={className}
    aria-live="assertive"
    role="alert"
    data-slnm-id={dataSlnmId ? `${dataSlnmId}-error` : ''}
    {...others}
  >
    {children}
  </Para>
);

FieldError.defaultProps = {
  dataSlnmId: '',
  inheritedStyles: '',
  className: '',
};

export default styled(FieldError)`
  ${styles};
`;

export { FieldError as FieldErrorVanilla };
