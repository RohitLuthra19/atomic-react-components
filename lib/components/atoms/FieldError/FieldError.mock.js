const defaultMessage = 'This input field\'s error message';

export default defaultMessage;
