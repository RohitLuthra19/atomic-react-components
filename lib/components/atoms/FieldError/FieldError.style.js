import { css } from 'styled-components';

const styles = css`
  font-family: ${props => props.theme.fontFamilySecondary};
  font-size: 12px;
  text-align: left;
  color: ${props => props.theme.textColorError};
  margin: 0;

  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default styles;
