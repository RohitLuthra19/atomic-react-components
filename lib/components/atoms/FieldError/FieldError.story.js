// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import defaultMessage from './FieldError.mock';
import FieldError, { FieldErrorVanilla } from '.';

storiesOf('Atoms', module).addWithChapters('FieldError', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <FieldErrorVanilla {...defaultMessage} className="hide-default-sample" />
          ),
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Input Field Error Message',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <FieldError {...defaultMessage} />,
        },
      ],
    },
  ],
});
