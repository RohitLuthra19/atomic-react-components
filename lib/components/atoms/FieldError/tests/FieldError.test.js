import React from 'react';
import { shallow, render } from 'enzyme';
import { FieldErrorVanilla } from '../FieldError';

describe('FieldError Component', () => {
  let fieldErrorComponent = '';
  beforeEach(() => {
    fieldErrorComponent = shallow(<FieldErrorVanilla>Some Error Occurred</FieldErrorVanilla>);
  });

  test('should render correctly', () => {
    expect(fieldErrorComponent).toMatchSnapshot();
  });
  test('should check the data attribute of field error', () => {
    const sampleProps = { elementLocator: 'my-custom-field-error-locator' };
    const fieldError = render(<FieldErrorVanilla {...sampleProps} />);

    expect(fieldError.attr('data-locator')).toEqual(sampleProps.elementLocator);
  });
});
