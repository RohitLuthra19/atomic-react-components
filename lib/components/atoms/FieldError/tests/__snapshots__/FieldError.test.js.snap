// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldError Component should render correctly 1`] = `
<Para
  aria-live="assertive"
  className=""
  data-locator="field-error-some-error"
  data-slnm-id=""
  role="alert"
>
  Some Error Occurred
</Para>
`;
